homeassistant/components/matter/__init__.py

```diff
@@ -0,0 +1,234 @@
+    """Init services."""
+
+    async def commission(call: ServiceCall) -> None:
+        """Handle commissioning."""
+        matter: Matter = list(hass.data[DOMAIN].values())[0]
+        try:
+            await matter.commission(call.data["code"])
+        except FailedCommand as err:
+            raise HomeAssistantError(str(err)) from err
+
+    async_register_admin_service(
+        hass,
+        DOMAIN,
+        "commission",
+        commission,
+        vol.Schema({"code": str}),
+    )
+
+    async def accept_shared_device(call: ServiceCall) -> None:
+        """Accept a shared device."""
+        matter: Matter = list(hass.data[DOMAIN].values())[0]
+        try:
+            await matter.commission_on_network(call.data["pin"])
+        except FailedCommand as err:
+            raise HomeAssistantError(str(err)) from err
+
+    async_register_admin_service(
+        hass,
+        DOMAIN,
+        "accept_shared_device",
+        accept_shared_device,
+        vol.Schema({"pin": vol.Coerce(int)}),
+    )
+
+    async def set_wifi(call: ServiceCall) -> None:
+        """Handle set wifi creds."""
+        matter: Matter = list(hass.data[DOMAIN].values())[0]
+        try:
+            await matter.client.driver.device_controller.set_wifi_credentials(
+                call.data["network_name"], call.data["password"]
+            )
+        except FailedCommand as err:
+            raise HomeAssistantError(str(err)) from err
+
+    async_register_admin_service(
+        hass,
+        DOMAIN,
+        "set_wifi",
+        set_wifi,
+        vol.Schema(
+            {
+                "network_name": str,
+                "password": str,
+            }
+        ),
+    )
+
+    async def set_thread(call: ServiceCall) -> None:
+        """Handle set Thread creds."""
+        matter: Matter = list(hass.data[DOMAIN].values())[0]
+        thread_dataset = bytes.fromhex(call.data["thread_operation_dataset"])
+        try:
+            await matter.client.driver.device_controller.set_thread_operational_dataset(
+                thread_dataset
+            )
+        except FailedCommand as err:
+            raise HomeAssistantError(str(err)) from err
+
+    async_register_admin_service(
+        hass,
+        DOMAIN,
+        "set_thread",
+        set_thread,
+        vol.Schema({"thread_operation_dataset": str}),
+    )
+
+    @callback
+    def _node_id_from_ha_device_id(ha_device_id: str) -> int | None:
+        """Get node id from ha device id."""
+        dev_reg = dr.async_get(hass)
+        device = dev_reg.async_get(ha_device_id)
+
+        if device is None:
+            return None
+
+        matter_iden = [iden for iden in device.identifiers if iden[0] == DOMAIN]
+
+        if not matter_iden:
+            return None
+
+        unique_id = matter_iden[0][1]
+
+        matter: Matter = list(hass.data[DOMAIN].values())[0]
+
+        # This could be more efficient
+        for node in matter.get_nodes():
```

Comment:
can't we create a mapping in the client lib so we can get a node by id and/or unique_id ?
Or maybe ask ourselves why we used the node_id in the first place if we prefer the unique_id.
It's not that big of a deal but if we're going to use this helper a lot it needs to be performant.